### PR TITLE
Add PacketBatch packet_indexes stat

### PR DESCRIPTION
**Pull request by buffalu**
#### Problem
Suspect duplicates are screwing with max batch size for parallel execution.

#### Summary of Changes
Add metrics to collect data on packet_indexes.

Fixes #

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -734,7 +759,7 @@ impl BankingStage {
         let recorder = poh_recorder.lock().unwrap().recorder();
         let socket = UdpSocket::bind("0.0.0.0:0").unwrap();
         let mut buffered_packet_batches = VecDeque::with_capacity(batch_limit);
-        let banking_stage_stats = BankingStageStats::new(id);
+        let mut banking_stage_stats = BankingStageStats::new(id);
```
Comment (buffalu) [inline core/src/banking_stage.rs:762]: still learning rust, a little confused on why we don't need this for atomicXYZ but we do for histogram

**Review by segfaultdoc — APPROVED** [review]

File: core/src/banking_stage.rs (hunk 1 above)
Comment (brooksprumo) [reply core/src/banking_stage.rs:762]: Atomics have interior mutability, so they are not declared as `mut`.

Comment (buffalu) [reply core/src/banking_stage.rs:762]: ok thanks for sharing! let me know if you'd prefer me to change it such that mut isn't required!

Comment (brooksprumo) [reply core/src/banking_stage.rs:762]: You'll want to make sure that `cargo check` and `cargo clippy` both pass. You can see that the CI "checks" failed, and you can see the output [here](https://buildkite.com/solana-labs/solana/builds/63583#dae208d0-329c-4716-ac3d-8769a6bdc3d5). This log also includes the commands that are run, so you can mimic them locally to ensure this CI step passes.

File: core/src/banking_stage.rs [hunk 2]
```diff
@@ -95,6 +96,7 @@ pub struct BankingStageStats {
     current_buffered_packet_batches_count: AtomicUsize,
     rebuffered_packets_count: AtomicUsize,
     consumed_buffered_packets_count: AtomicUsize,
+    packet_batch_indices_len: Histogram,
```
Comment (buffalu) [inline core/src/banking_stage.rs:99]: indexes vs indices? :)

Comment (t-nelson) [reply core/src/banking_stage.rs:99]: "indexes" would be consistent with the name of the thing that its measuring.  then again, already cleared CI, so...  up to you!

Comment (buffalu) [reply core/src/banking_stage.rs:99]: ok game time decision, going to switch it up real quick

**Pull request merged**